### PR TITLE
plugin, privilege: fix err check

### PR DESCRIPTION
**Pull request by Tjianke**
### What problem does this PR solve?

Issue Number: #22979  

Problem Summary:

### What is changed and how it works?

What's Changed: added err check

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- No release note

### CODE REVIEW

File: privilege/privileges/privileges_test.go [hunk 1]
```diff
@@ -573,7 +573,10 @@ func (s *testPrivilegeSuite) TestCheckCertBasedAuth(c *C) {
 		},
 		tls.TLS_AES_128_GCM_SHA256, func(c *x509.Certificate) {
 			var url url.URL
-			url.UnmarshalBinary([]byte("spiffe://mesh.pingcap.com/ns/timesh/sa/me1"))
+			err := url.UnmarshalBinary([]byte("spiffe://mesh.pingcap.com/ns/timesh/sa/me1"))
+			if err != nil {
+				panic(err)
```
Comment (tangenta) [inline privilege/privileges/privileges_test.go:578]: Please always prefer to use `c.Assert`.

File: plugin/spi_test.go [hunk 2]
```diff
@@ -41,7 +41,10 @@ func TestExportManifest(t *testing.T) {
 		},
 	}
 	exported := plugin.ExportManifest(manifest)
-	exported.OnInit(context.Background(), exported)
+	err := exported.OnInit(context.Background(), exported)
+	if err != nil {
+		panic(err)
```
Comment (zz-jason) [inline plugin/spi_test.go:46]: could you use `t.Fatalf()`?

Comment (Tjianke) [reply plugin/spi_test.go]: Done.

**Review by xiongjiwei — APPROVED** [review]

**Pull request merged**